src/Common/precompiled.h

```diff
@@ -203,18 +220,52 @@ typedef union _LARGE_INTEGER {
     inline T& operator^= (T& a, T b) { return reinterpret_cast<T&>( reinterpret_cast<std::underlying_type<T>::type&>(a) ^= static_cast<std::underlying_type<T>::type>(b) ); }
 #endif
 
-// MEMPTR
-#include "Common/MemPtr.h"
+#if defined(_MSC_VER)
+    #define UNREACHABLE __assume(false)
+#elif defined(__GNUC__)
+    #define UNREACHABLE __builtin_unreachable()
+#else
+    #define UNREACHABLE
+#endif
 
-#define MPTR_NULL	(0)
+#if defined(_MSC_VER)
+    #define DEBUG_BREAK __debugbreak()
+#else
+    #include <csignal>
+    #define DEBUG_BREAK raise(SIGTRAP) 
+#endif
 
-// macros
-#if BOOST_OS_WINDOWS
-#define DLLEXPORT __declspec(dllexport)
+#if defined(_MSC_VER)
+    #define THREAD_LOCAL __declspec(thread)
+#elif defined(__GNUC__)
+    #define THREAD_LOCAL __thread
 #else
-#define DLLEXPORT __declspec(dllexport)
+    #define THREAD_LOCAL thread_local
 #endif
 
+#if defined(_MSC_VER)
+    #define DLLEXPORT __declspec(dllexport)
+    #define DLLIMPORT __declspec(dllimport)
+#elif defined(__GNUC__)
+    #define DLLEXPORT __attribute__((dllexport))
+    #define DLLIMPORT __attribute__((dllimport))
+#else
+    #error No definition for DLLEXPORT and DLLIMPORT
+#endif
+
+#if defined(_MSC_VER)
+    #define NOINLINE __declspec(noinline)
+#elif defined(__GNUC__)
+    #define NOINLINE __attribute__((noinline))
+#else
+    #error No definition for NOINLINE
+#endif
```

Comment:
NOINLINE can be removed, see other comment.

Reply:
Done.